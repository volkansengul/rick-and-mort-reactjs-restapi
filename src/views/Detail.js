import React from 'react';
import {Link, Redirect} from "react-router-dom";
import Card from '../components/CardList/Card';
import Episodes from "../components/Episodes";

export default class Detail extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            character: undefined
        };
        this.characterId = parseInt(props.match.params.characterId);
    }

    characterReady = (data) => {
        this.setState({
            character: data
        })
    }


    render() {
        const redirect = (this.characterId === undefined  || isNaN(this.characterId)) ?<Redirect to="/error" push/> : null;
        return (
            <div>
                {redirect}
                <div className={'go-back-container'}>
                    <Link to={'/'} className={'go-back'}>&laquo; Go Back</Link>
                </div>
                <div className={'clear'}/>
                <div className="container">
                    <Card characterId={this.characterId}
                          callback={this.characterReady}>
                        <Card.Details/>
                    </Card>
                    <div className={'episodes-container'}>
                        <Episodes
                            episodeIdList={this.state.character ? this.state.character.episode : null}
                            limit={5}/>
                    </div>
                </div>
            </div>

        )
    }
}