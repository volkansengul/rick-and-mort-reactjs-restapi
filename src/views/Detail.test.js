import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';

import Detail from './Detail';

describe('Character detail page', () => {
    it('renders without crashing', () => {
        const episodeList = renderer.create(
            <Router>
                <Detail match={{
                    params: {
                        characterId: 1
                    }
                }}/>
            </Router>
        ).toJSON();
        expect(episodeList).toMatchSnapshot();
    });
});