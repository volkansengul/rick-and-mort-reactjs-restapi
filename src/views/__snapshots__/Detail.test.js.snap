// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character detail page renders without crashing 1`] = `
<div>
  <div
    className="go-back-container"
  >
    <a
      className="go-back"
      href="/"
      onClick={[Function]}
    >
      Â« Go Back
    </a>
  </div>
  <div
    className="clear"
  />
  <div
    className="container"
  >
    <div
      className="loading"
    />
    <div
      className="episodes-container"
    >
      <div
        className="episodes"
      >
        <h3
          className="title"
        >
          Last 5 Episodes
        </h3>
      </div>
    </div>
  </div>
</div>
`;
